### PR TITLE
tests: Refactors agnhost image pod usage in tests

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

A previous commit added  a few agnhost related functions that creates agnhost pods / containers for general purposes.

Refactors tests to use those functions.

**Which issue(s) this PR fixes**:

Fixes #
Depends On: #87266

**Special notes for your reviewer**:

/cc @alejandrox1 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/framework/pod/resource.go [hunk 1]
```diff
@@ -409,49 +409,55 @@ func isNotRestartAlwaysMirrorPod(p *v1.Pod) bool {
...
-			GenerateName: generateName,
-			Namespace:    ns,
-		},
-		Spec: v1.PodSpec{
-			TerminationGracePeriodSeconds: &immediate,
-			Containers: []v1.Container{
-				{
-					Name:  "agnhost-pause",
-					Image: imageutils.GetE2EImage(imageutils.Agnhost),
-					Args:  []string{"pause"},
```
Comment (oomichi) [inline test/e2e/framework/pod/resource.go:450]: Don't we need to pass this "pause" as args of NewAgnhostPod()?

File: test/e2e/network/service.go [hunk 2]
```diff
@@ -3612,25 +3608,12 @@ func createPausePodDeployment(cs clientset.Interface, name, ns string, replicas
...
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   name,
-			Labels: labels,
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name:  "pause",
-					Image: imageutils.GetE2EImage(imageutils.Agnhost),
-					Args:  []string{"pause"},
```
Comment (oomichi) [inline test/e2e/network/service.go:3625]: ditto

File: test/e2e/node/pods.go [hunk 3]
```diff
@@ -61,23 +61,10 @@ var _ = SIGDescribe("Pods Extended", func() {
...
-						"name": "foo",
-						"time": value,
-					},
-				},
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						{
-							Name:  "agnhost",
-							Image: imageutils.GetE2EImage(imageutils.Agnhost),
-							Args:  []string{"pause"},
```
Comment (oomichi) [inline test/e2e/node/pods.go:77]: ditto

**Review by oomichi — COMMENTED** [review]
Looks good cleanup for me, just few comments.

/cc @oomichi

File: test/e2e/framework/pod/resource.go (hunk 1 above)
Comment (claudiubelu) [reply test/e2e/framework/pod/resource.go:450]: See above, if no argument is given for ``args...``, ``NetAgnhostContainer`` will consider this to be the default.

Comment (oomichi) [reply test/e2e/framework/pod/resource.go:450]: Ah, I got it.
Thanks for your explanation.

/lgtm

**Review by alejandrox1 — COMMENTED** [review]
/approve
/lgtm
/priority backlog

Thank you @claudiubelu !

**Pull request merged**